### ISSUE TITLE
Aria compatibility with Modal, pushed pages, Action Sheet and Alert

### ISSUE THREAD

**Issue by mnogueron** [ionic-team/ionic#7092]
#### Short description of the problem:

**Only tested on Android**
1- When displaying a modal, a new page with nav.push(), an ActionSheet or an Alert, the parent page is still read by Talkback.

Here the parent content is focused and read by Talkback even if the modal is in foreground:
![device-2016-06-28-111731](https://cloud.githubusercontent.com/assets/8511318/16410263/faf9739c-3d21-11e6-88ca-0e0e046689fe.png)

2- When displaying a modal or a new page with nav.push() the position of the parent's navbar is still visible by Talkback but not read. Due to that, all the modal's content is shifted.

Here the close button is focused, but the focus is not made on the right place:
![device-2016-06-28-112505](https://cloud.githubusercontent.com/assets/8511318/16410793/6f5ea39a-3d24-11e6-9869-9bd6e52a95dd.png)

3- When displaying an ActionSheet all the content is only read and not focused by Talkback.

The Share field is actually selected but there is no focus:
![device-2016-06-28-114326](https://cloud.githubusercontent.com/assets/8511318/16411059/9785d252-3d25-11e6-853b-5d1a0e214739.png)

4- When displaying an Alert, the first button is focused but not the title or the subtitle.
When an input field, a radio button or a checkbox is displayed, the field is focused but not the title or the subtitle.

**Steps to reproduce:**
1. Enable Talkback
2. Run demo for each components 

**Which Ionic Version?**
ionic: 2.0.0-beta.31
ionic-angular: 2.0.0-beta.9
#### Your system information:

Cordova CLI: 6.1.1
Gulp version:  CLI version 3.9.1
Gulp local:   Local version 3.9.1
Ionic Framework Version: 2.0.0-beta.9
Ionic CLI Version: 2.0.0-beta.31
Ionic App Lib Version: 2.0.0-beta.17
OS: Distributor ID: Ubuntu Description: Ubuntu 14.04.4 LTS 
Node Version: v6.2.0

**Comment by mnogueron** [comment]
Still the same issues on beta.10

**Comment by mnogueron** [comment]
It seems that animations are causing the bad positioning of the focus for a pushed page, modal alert or action sheet. Talkback seems to initiate before the end of the animations that causes the bad positioning of each element in the page.

A working workaround consists of adding this to your pushed or modal page:

```
ionViewWillEnter(){
	document.body.style.width = "99.99999%";
}

ionViewDidEnter() {
       document.body.style.removeProperty("width");
}
```
By setting the width of the body to a different value than 100%, it will force Talkback to recalculate the position of the focus, and removing width after will do nothing but setting it back to 100% (Talkback seems to not recalculate position on style removed).

It looks like that's the same thing for alert and actionsheet, but the workaround is not very clean for these elements.
